lib/src/merge.rs

```diff
@@ -0,0 +1,127 @@
+// Copyright 2023 The Jujutsu Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+use std::hash::Hash;
+
+use itertools::Itertools;
+
+/// Attempt to resolve trivial conflicts between the inputs. Even indexes are
+/// positive terms, and odd indexes are negative terms. The length must be odd.
+pub fn trivial_merge<T>(input: &[T]) -> Option<T>
```

Comment:
This signature is pretty convenient for the 3-way merge case, but it's a bit odd in every other case. I wonder if it would be simpler to accept adds and removes as different parameters. This function body would be a bit simpler, and maybe some of `files.rs` would be too.

Reply:
I had considered making the signature be `trivial_merge<T>(base: T, diffs: &[(T, T)])` so we can remove the assertion about the length, but it didn't seem worth it to me. I had not considered two separate lists, but that might make sense. Maybe it makes the diffs less clear because they're not right next to each other in a single list, but that's probably no big deal. I've added a commit on top to show what it looks like. I think it looks good. If you and @ilyagr agree, I'll squash it into the commit that introduced this function.

Btw, this PR is also very much related to PR #1480. In fact, @ilyagr's proposal there about always storing conflicts at the commit level is why I wrote this PR - I started working on it and felt like we want to do multi-way tree merges instead of doing repeated 3-way merges (now I don't even remember why I felt that way).

Reply:
Thanks! I like the new signature a lot more. I think it's a lot easier to read without having to repeat `2` all around. I was surprised to find the `trivial_merge()` test cases easier to read too - I previously had a hard time keeping track of the adds/removes.

Reply:
I also like it better :) I moved it to the first commit in this PR. I also swapped the `adds` and `removes` arguments, putting `removes` first.